server/dataBase/db_build.sql

```diff
@@ -0,0 +1,55 @@
+BEGIN ;
+DROP TABLE IF EXISTS girl, company, work_experience, school, registeration CASCADE;
+
+CREATE TABLE school (
+id SERIAL PRIMARY KEY,
+name VARCHAR(255)
+);
+
+CREATE TABLE girl (
+id SERIAL PRIMARY KEY,
+user_name VARCHAR(50) NOT NULL,
+first_name VARCHAR(50) NOT NULL,
+last_name VARCHAR(50) NOT NULL,
+email VARCHAR NOT NULL,
+school_id INTEGER REFERENCES school(id) ON DELETE CASCADE,
+birthDate DATE,
+password TEXT
+);
+
+CREATE TABLE company (
+id SERIAL PRIMARY KEY,
+logo TEXT,
+company_name VARCHAR,
+description TEXT,
+mobile VARCHAR,
+website VARCHAR,
+email VARCHAR,
+facebook VARCHAR
+);
+
+CREATE TABLE work_experience (
+id SERIAL PRIMARY KEY,
+company_id INTEGER REFERENCES company(id) ON DELETE CASCADE,
+location VARCHAR(50),
+subject VARCHAR(50),
+post_code INTEGER,
+start_date DATE,
+end_date DATE,
+pocket_money INTEGER,
+title VARCHAR,
+max_number INTEGER,
+description TEXT,
+video TEXT,
+start_time VARCHAR,
+end_time VARCHAR
+);
+
+CREATE TABLE registeration (
+id SERIAL PRIMARY KEY,
+work_experience_id INTEGER REFERENCES work_experience(id) ON DELETE CASCADE,
+girl_id INTEGER REFERENCES girl(id) ON DELETE CASCADE,
+accepted BOOLEAN
+);
```

Comment:
What do you think about adding a `registration_date` to this table, so the frontend can show how many days ago someone applied to a job?

Another idea I had is to change `accepted BOOLEAN` into `status VARCHAR(50)` which would give you more control - a registration's status could then be "waiting" (company has not responded yet), "accepted", or "rejected".  It's not totally necessary but might be nice to show in the UI, and you also don't have to change the UI design right now, but changing the schema now would give you the freedom to make that UI change in the future.  

And one last idea, related to those - adding an `explanation TEXT` attribute so companies could write a message to the girl when they reject an application (for example: "sorry, we have filled this position but check back with us next month for more openings!")

Reply:
You are absolutely right about adding those fields, and I think we planned for them on paper. Anyway, this is not final, because we decided to use Sequelize.